cmake/platform/windows.cmake

```diff
@@ -88,3 +88,28 @@ if(MSVC)
   # https://blogs.msdn.microsoft.com/vcblog/2018/04/09/msvc-now-correctly-reports-__cplusplus/
   add_compile_options(/Zc:__cplusplus)
 endif()
+
+#
+# Helper macros added to the catkin system to replace Windows lack of shebang support.
+#
+# Code courtesy of Yujin Robot.  Derived from https://github.com/ros-windows/win_ros
+#
+
+# distutils does not have support for 'entry_points'
+macro(add_windows_python_batch_helper name)
+  set(PYTHON_TARGET ${name})
+  configure_file(${catkin_EXTRAS_DIR}/templates/python_wrapper.bat.in
+                 ${CATKIN_DEVEL_PREFIX}/bin/${name}.bat)
+  install(PROGRAMS ${CATKIN_DEVEL_PREFIX}/bin/${PYTHON_TARGET}.bat 
+          DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION})
+endmacro()
+
+# some python scripts need an executable, not a batch file
+macro(add_windows_python_exe_helper name)
+  # make the target name unique so we don't clash (e.g. rosbag target)
+  add_executable(${name}_exe ${catkin_EXTRAS_DIR}/templates/ros_bin.cpp)
```

Comment:
The function should not decide on the target name internally. That has bitten us in various other API. Please create an argument for it which the caller needs to provide.

Reply:
thanks for sharing the experience with us! taking some time to rework on this change

Reply:
Do you have an estimate for an updated patch since I am considering to do a new release of the package soon?

Reply:
working on `catkin` and `ros_core` packages right now, I will send out an update before EOD 2/7. should I ping on this thread when that happens?

Reply:
Great. It is always good to comment on a PR when pushing new commits to ensure others get a notification email.

Reply:
add `target` parameter